lib/utils.js

```diff
@@ -81,6 +87,34 @@ function formatTime (epoch, translateTime = false) {
   return dateformat(instant, `UTC:${translateTime}`)
 }
 
+/**
+ * Replacement for the Date() constuctor. The argument may be any argument
+ * that is valid for the Date constuctor, or an epoch provided as a string.
+ *
+ * @epoch {string|number} - Argument for the date constructor.
+ *
+ * @returns {Date} date
+ */
+function createDate (epoch) {
+  // If epoch is already a valid argument, return the valid Date
+  let date = new Date(epoch)
+  if (isValidDate(date)) {
+    return date
+  }
+
+  // Convert to a number to permit epoch as a string
+  date = new Date(+epoch)
+  return date
+}
+
+/**
+ * Returns true if the argument is a Date object and not 'Invalid Date'
+ * @param {Date} date
+ */
+function isValidDate (date) {
```

Comment:
This function should be added to `module.exports.internals` and tested explicitly in `test/lib/utils.internals.test.js`.